src/app/components/home/home.component.ts

```diff
@@ -26,15 +26,39 @@ export class HomeComponent implements OnInit, AfterViewInit, OnDestroy {
   iterationIds: number[] = [0, 0]; // Initialization is a temporary fix for bug in carousel's looping
   @ViewChild(CarouselComponent) carousel: CarouselComponent;
 
+  private initSubject = new Subject<void>();
+  private updateSubject = new Subject<void>();
   private dataSubscription: Subscription;
+  private routerSubscription: Subscription;
 
   constructor(
     public readonly modalService: DescriptionModalService,
     private readonly changeDetector: ChangeDetectorRef,
     private readonly dataService: MacroscopeDataService,
-    private readonly route: ActivatedRoute,
-    private readonly router: Router
-  ) { }
+    private readonly router: Router,
+    route: ActivatedRoute
+  ) {
+    const { initSubject, updateSubject } = this;
```

Comment:
spacing looks off here

Reply:
It is not

Reply:
line 41 needs to be indented more

Reply:
and the following lines too I think

Reply:
No, the closing parenthesis should have the same indentation as 'constructor'. The body is the indented once.